content/blog/announcing-enum-support/index.md

```diff
@@ -0,0 +1,285 @@
+
+If we look at the [resource docs](https://www.pulumi.com/docs/reference/pkg/aws/s3/bucket/#acl_nodejs), we can see that the `acl` property can only be set to one of a few different values: `private`, `public-read`, `public-read-write`, `aws-exec-read`, `authenticated-read`, and `log-delivery-write`. The `acl` property is the perfect candidate for an enum type and is emitted as one, so you can use the following code instead.
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import { s3 } from "@pulumi/aws";
+
+const myBucket = new s3.Bucket("myBucket", { acl: s3.CannedAcl.Private });
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+import pulumi
+from pulumi_aws import s3
+
+my_bucket = s3.Bucket("myBucket", acl=s3.CannedAcl.PRIVATE)
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+using Pulumi;
+using S3 = Pulumi.Aws.S3;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var myBucket = new S3.Bucket("myBucket", new S3.BucketArgs
+        {
+            Acl = S3.CannedAcl.Private,
+        });
+    }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        myBucket, err := s3.NewBucket(ctx, "myBucket", &s3.BucketArgs{
+            Acl: s3.CannedAclPrivate,
+        })
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+## IDE Superpowers
+
+Enum properties provide discoverable and normalized constants that can be used in place of raw strings, allowing you to move faster with the added boosts of IDE tooling like type hints and autocomplete.
+
+Using the provided constants, you can avoid referring back to the documentation to remember the valid values and save precious moments in the development cycle that would be lost to debugging errors caused by typos.
+
+{{< chooser >}}
+{{< choosable language typescript >}}
+
+![ENUM_TYPESCRIPT](ts-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+![ENUM_PYTHON](python-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+![ENUM_CSHARP](csharp-enum.gif)
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+![ENUM_GO](go-enum.gif)
+
+{{< /choosable >}}
+{{< /chooser >}}
+
+## Optimized for flexibility
+
+While some properties make sense as "strict" enums (i.e., the input value **must** be one of the enumerated values), there are times when it would be beneficial to have constants for commonly used values without restricting the input to **only** those values. We call this second category "relaxed" enums, and we model them a little differently in our schema.
```

Comment:
The schema is quite an "internal implementation detail".  The interesting question for users is more how this difference maps to how these are represented in each of the languages they will use - right?

Reply:
Yeah, I get that. I went down the path of writing out the representation in each of the languages but it was getting super caveat-y (In Go, relaxed enums are not represented any differently than regular ol' string types). I decided to just go with the json representation because it was feeling a little bit like there were too many things that may need explanations. I can go back and redo it if you feel like that would really help.

Reply:
I see. If you think this is better than what we could do with language-based examples - definitely happy to leave as is here.

Reply:
Heh, I actually went ahead and added the language-based examples in the latest commit. Let me know if you think it's better. I do still need to add some sort of explainer for Go re: relaxed enums. Trying to figure out the balance of "enough information to make sense but not too much to overwhelm"